### PR TITLE
FSA22V2-82: Add support for fetching all heros

### PR DESCRIPTION
**Pull request by marissahuysentruyt**
### Description

Create a function `getCharactersAll` that returns the entire Sparkhero API as an array. Additionally, use that array to determine how many items were returned, and insert that into a sneaky `console.debug` message. 

2 additional tests were written in the `characterServices.spec.js` file to confirm...
1.  that `getCharactersAll` does indeed return an array, and
2. that an error message is thrown if `getCharactersAll` can't complete. 

### Spec

See Story: [FSA22V2-82](https://sparkbox.atlassian.net/jira/software/c/projects/FSA22V2/boards/125?modal=detail&selectedIssue=FSA22V2-82)

### Validation

* [x] This PR has code changes, and our linters still pass.
* [x] This PR affects production code, so it was browser tested (see below).
* [x] This PR has new code, so new tests were added or updated, and they pass.
* [x] The content of this PR requires documentation, so we added a detailed description of the component's purpose, requirements, quirks, and instructions for use by designers and developers. Along with accessibility information if pertinent.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
3. Pull down all related branches.
4. Confirm all tests pass.

---

#### Gold Level Support

In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

* [x] Safari (last 2 major versions)
* [x] Chrome (last 6 months)
* [x] Firefox (last 6 months)

**Windows**

* [ ] IE 11+
* [x] Chrome (last 6 months)
* [x] Firefox (last 6 months)
* [x] Edge 18

**Mobile**

* [x] Safari (last 2 major versions)
* [x] Android 7+
* [x] Samsung Browser (last year)
* [x] Chrome (last 6 months)
* [x] Firefox (last 6 months)

### CODE REVIEW

**Comment by marissahuysentruyt** [comment]
I am not done browser testing everything.

File: src/js/services/characterService.spec.js [hunk 1]
```diff
@@ -26,6 +26,39 @@ describe('Character Service', () => {
...
+      beforeAll(async () => {
+        get.mockImplementation(() => { throw new Error(MOCK_ERROR); });
+        character = await getCharactersAll();
+      });
+      it('returns an empty array when errors are thrown', () => {
+        expect(character).toMatchObject([]);
+      });
+      it('logs an error message when an error is thrown', async () => {
+        expect(console.error).toHaveBeenCalledWith(expect.stringContaining(MOCK_ERROR));
+      });
```
Comment (QuinnrElder) [inline src/js/services/characterService.spec.js:58]: Testing HAPPY and SAD paths is always a great practice! Nice work!!

File: src/js/services/characterService.spec.js [hunk 2]
```diff
@@ -26,6 +26,39 @@ describe('Character Service', () => {
...
+
+        const character = await getCharactersAll();
+        expect(character).toEqual(expect.arrayContaining(MOCK_CHARACTER_ARRAY));
+      });
+    });
+
+    // getCharactersAll wasn't able to access the API, so got an error message
+    describe('unsuccessful API call', () => {
+      let character;
+      const MOCK_ERROR = 'Unable to get all characters!';
```
Comment (QuinnrElder) [inline src/js/services/characterService.spec.js:48]: Nitpick ~ maybe think of updating ```Character``` to ```Characters``` ~ when naming variables try and think of the content that they hold. In this case, the value of ```Character``` is an array(list) of data pointing to a plural naming convention. 😇

#naming-is-hard

Comment (marissahuysentruyt) [reply src/js/services/characterService.spec.js:48]: That makes total sense. I updated it!

**Review by QuinnrElder — APPROVED** [review]
Love how thorough you were in your PR description! LGTM ~ nice job!

**Comment by QuinnrElder** [comment]
> Technically...I don't think this passes the IE11 check. 🫢

```IE``` is no longer supported so I believe that we no longer need to test against this ~ can confirm with your instructors!

**Pull request merged**